nistats/datasets.py

```diff
@@ -334,69 +485,134 @@ def fetch_spm_auditory(data_dir=None, data_name='spm_auditory',
+        
+
+    # glob for anat data
+    _subject_data = _get_anatomical_data_spm_multimodal(subject_dir, _subject_data)
+    if not _subject_data:
+        return None
+    
+    return Bunch(**_subject_data)
+
+
+def _download_data_spm_multimodal(data_dir, subject_dir, subject_id):
+    print("Data absent, downloading...")
+    urls = [
+        # fmri
+        ("http://www.fil.ion.ucl.ac.uk/spm/download/data/mmfaces/"
+        "multimodal_fmri.zip"),
 
-        subject_data = {}
-        for file_name in SPM_AUDITORY_DATA_FILES:
-            file_path = os.path.join(subject_dir, file_name)
-            if os.path.exists(file_path):
-                subject_data[file_name] = file_path
-            else:
-                print("%s missing from filelist!" % file_name)
-                return None
-
-        _subject_data = {}
-        _subject_data["func"] = sorted(
-            [subject_data[x] for x in subject_data.keys()
-             if re.match("^fM00223_0\d\d\.img$", os.path.basename(x))])
-
-        # volumes for this dataset of shape (64, 64, 64, 1); let's fix this
-        for x in _subject_data["func"]:
-            vol = nib.load(x)
-            if len(vol.shape) == 4:
-                vol = nib.Nifti1Image(vol.get_data()[:, :, :, 0],
-                                      vol.affine)
-                nib.save(vol, x)
+        # structural
+        ("http://www.fil.ion.ucl.ac.uk/spm/download/data/mmfaces/"
+         "multimodal_smri.zip")
+        ]
 
-        _subject_data["anat"] = [subject_data[x] for x in subject_data.keys()
-                                 if re.match("^sM00223_002\.img$",
-                                             os.path.basename(x))][0]
+    for url in urls:
+        archive_path = os.path.join(subject_dir, os.path.basename(url))
+        _fetch_file(url, subject_dir)
+        try:
+            _uncompress_file(archive_path)
+        except:
+            print("Archive corrupted, trying to download it again.")
+            return fetch_spm_multimodal_fmri(data_dir=data_dir,
+                                             data_name="",
+                                             subject_id=subject_id)
 
-        # ... same thing for anat
-        vol = nib.load(_subject_data["anat"])
-        if len(vol.shape) == 4:
-            vol = nib.Nifti1Image(vol.get_data()[:, :, :, 0],
-                                  vol.affine)
-            nib.save(vol, _subject_data["anat"])
+    return _glob_spm_multimodal_fmri_data(subject_dir)
 
-        return Bunch(**_subject_data)
 
-    # maybe data_dir already contains the data ?
-    data = _glob_spm_auditory_data()
-    if data is not None:
-        return data
+def _make_events_filepath_spm_multimodal_fmri(_subject_data, session):
+    key = 'trials_ses{}'.format(session+1)
+    events_file_location = os.path.dirname(_subject_data[key])
+    events_filename = 'session{}_events.tsv'.format(session+1)
```

Comment:
again

Reply:
In standalone expressions, not in function/method args

Reply:
Regardless, I'll obviate this concern soon